src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1554,31 +1554,38 @@ LogicalResult ONNXBatchNormalizationInferenceModeOp::inferShapes(
 // Reshape
 //===----------------------------------------------------------------------===//
 
-LogicalResult ONNXReshapeOp::inferShapes(
-    std::function<void(mlir::Region &)> doShapeInference) {
+static LogicalResult verify(ONNXReshapeOp op) {
   // Cannot infer shape if no shape tensor is specified.
-  if (!data().getType().isa<RankedTensorType>())
+  if (!op.data().getType().isa<RankedTensorType>())
```

Comment:
This call is a bit more precise.

`if (! hasShapeAndRank(op.data())`

please convert elsewhere in the verifiers that you added

Reply:
Done